lib/pages/chat/chat_room_search_mixin.dart

```diff
@@ -97,10 +103,17 @@ mixin ChatRoomSearchMixin {
     canGoUp.value = false;
   }
 
-  void _scrollToEvent(Either<Failure, Success> event) {
+  Future _scrollToEvent(Either<Failure, Success> event) async {
     final index = event.getSuccessOrNull<ChatRoomSearchSuccess>()?.eventIndex;
     if (index != null) {
-      _scrollToIndex?.call(index);
+      _scrollingToIndexCount++;
+      try {
+        await _scrollToIndex?.call(index);
+      } catch (e) {
+        Logs().e('ChatRoomSearchMixin::_scrollToEvent $e');
+      } finally {
+        _scrollingToIndexCount--;
```

Comment:
can you explain why you increment index then decrement ?

Reply:
while user press next/previous search very quickly
it will call scroll to index multi time
_scrollingToIndexCount make sure all scrollToIndex completed